impl.go

```diff
@@ -118,7 +118,7 @@ func typeSpec(path string, id string, srcDir string) (Pkg, Spec, error) {
 	}
 
 	fset := token.NewFileSet() // share one fset across the whole package
-	for _, file := range pkg.GoFiles {
+	for _, file := range append(pkg.GoFiles, pkg.CgoFiles...) {
```

Comment:
I don't know offhand whether pkg.GoFiles is safe to append to. (Is there other stuff later on in that slice? Does anyone care if we clobber it?) I'd rather do something more verbose like

```go
var files []string
files = append(files, pkg.GoFiles...)
files = append(files, pkg.CgoFiles...)
```

But if you really want to be terse and inline, please do

`append(pkg.GoFiles[:len(pkg.GoFiles):len(pkg.GoFiles)], pkg.CgoFiles...)`

(Eventually impl should probably start using go/packages...)